### PR TITLE
added section about parallelism to delegation

### PR DESCRIPTION
**Pull request by bcoca**
also updated note about connection vars

##### ISSUE TYPE

- Docs Pull Request

##### COMPONENT NAME

docs

### CODE REVIEW

File: docs/docsite/rst/user_guide/playbooks_delegation.rst [hunk 1]
```diff
@@ -77,7 +77,26 @@ To specify more arguments, use the following syntax::
...
+By default Ansbile tasks are executed in parallel, delegating a task does not change this and does not handle concurrency issues (mulitple forks writing to same file).
+Most commonly users are affected by this when updating a single file on a single delegated to host for all hosts (via copy/template/lineinfile/etc), they will still operate in parallel forks (default 5) and overwrite each other.
+
+This can be handled in several ways::
+
+    ---
+    - name: handle concurrency via a loop on the hosts with `run_once: true`:
+      lineinfile: ...
+      run_once: true
+      loop: '{{ q("inventory_hostnames" ...}}'
```
Comment (mattclay) [inline docs/docsite/rst/user_guide/playbooks_delegation.rst]: The example block needs to be valid YAML + Jinja2 syntax:

```suggestion
      loop: '{{ q("inventory_hostnames") }}'
```

File: docs/docsite/rst/user_guide/playbooks_delegation.rst [hunk 2]
```diff
@@ -77,7 +77,24 @@ To specify more arguments, use the following syntax::
...
 
-The `ansible_host` variable reflects the host a task is delegated to.
+.. note::
+    - The `ansible_host` variable and other connection variables, if present, will reflect information of the host a task is delegated to, not the inventory_hostname.
+
+.. _delegate_parallel:
+
+Parallelism
+-----------
+By default Ansbile tasks are executed in parallel, delegating a task does not change this and does not handle concurrency issues (mulitple forks writing to same file).
```
Comment (Shrews) [inline docs/docsite/rst/user_guide/playbooks_delegation.rst]:
```suggestion
By default Ansible tasks are executed in parallel. Delegating a task does not change this and does not handle concurrency issues (multiple forks writing to same file).
```

File: docs/docsite/rst/user_guide/playbooks_delegation.rst [hunk 3]
```diff
@@ -77,7 +77,24 @@ To specify more arguments, use the following syntax::
...
+Most commonly users are affected by this when updating a single file on a single delegated to host for all hosts (via copy/template/lineinfile/etc), they will still operate in parallel forks (default 5) and overwrite each other.
+
+This can be handled in several ways::
+
+    - name: "handle concurrency via a loop on the hosts with `run_once: true`"
+      lineinfile: "<options here>"
+      run_once: true
+      loop: '{{ ansible_play_hosts_all }}'
+
+By using an intermidate play with  `serial: 1` or using  `throttle: 1` at task level.
```
Comment (Shrews) [inline docs/docsite/rst/user_guide/playbooks_delegation.rst]:
```suggestion
By using an intermediate play with  `serial: 1` or using  `throttle: 1` at task level.
```

File: docs/docsite/rst/user_guide/playbooks_delegation.rst [hunk 4]
```diff
@@ -77,7 +77,24 @@ To specify more arguments, use the following syntax::
             body: "{{ mail_body }}"
           run_once: True
 
-The `ansible_host` variable reflects the host a task is delegated to.
+.. note::
+    - The `ansible_host` variable and other connection variables, if present, will reflect information of the host a task is delegated to, not the inventory_hostname.
```
Comment (samccann) [inline docs/docsite/rst/user_guide/playbooks_delegation.rst]:
```suggestion
    - The `ansible_host` variable and other connection variables, if present, will reflect information about the host a task is delegated to, not the inventory_hostname.
```

File: docs/docsite/rst/user_guide/playbooks_delegation.rst [hunk 5]
```diff
@@ -77,7 +77,24 @@ To specify more arguments, use the following syntax::
...
+
+.. _delegate_parallel:
+
+Parallelism
+-----------
+By default Ansible tasks are executed in parallel. Delegating a task does not change this and does not handle concurrency issues (multiple forks writing to same file).
```
Comment (samccann) [inline docs/docsite/rst/user_guide/playbooks_delegation.rst]:
```suggestion
By default Ansible tasks are executed in parallel. Delegating a task does not change this and does not handle concurrency issues (multiple forks writing to the same file).
```

File: docs/docsite/rst/user_guide/playbooks_delegation.rst [hunk 6]
```diff
@@ -77,7 +77,24 @@ To specify more arguments, use the following syntax::
...
+Most commonly users are affected by this when updating a single file on a single delegated to host for all hosts (via copy/template/lineinfile/etc), they will still operate in parallel forks (default 5) and overwrite each other.
```
Comment (samccann) [inline docs/docsite/rst/user_guide/playbooks_delegation.rst]:
```suggestion
Most commonly, users are affected by this when updating a single file on a single delegated to host for all hosts (using the ``copy``, ``template``, or ``lineinfile`` modules, for example). They will still operate in parallel forks (default 5) and overwrite each other.
```

File: docs/docsite/rst/user_guide/playbooks_delegation.rst (hunk 4 above)
Comment (acozine) [inline docs/docsite/rst/user_guide/playbooks_delegation.rst]: Per our style guide, avoid future tense.

```suggestion
    - The `ansible_host` variable and other connection variables, if present, reflects information of the host a task is delegated to, not the inventory_hostname.
```

File: docs/docsite/rst/user_guide/playbooks_delegation.rst [hunk 7]
```diff
@@ -77,7 +77,24 @@ To specify more arguments, use the following syntax::
...
+
+This can be handled in several ways::
+
+    - name: "handle concurrency via a loop on the hosts with `run_once: true`"
```
Comment (samccann) [inline docs/docsite/rst/user_guide/playbooks_delegation.rst]:
```suggestion
    - name: "handle concurrency with a loop on the hosts with `run_once: true`"
```

File: docs/docsite/rst/user_guide/playbooks_delegation.rst [hunk 8]
```diff
@@ -77,7 +77,24 @@ To specify more arguments, use the following syntax::
...
           run_once: True
 
-The `ansible_host` variable reflects the host a task is delegated to.
+.. note::
+    - The `ansible_host` variable and other connection variables, if present, will reflect information of the host a task is delegated to, not the inventory_hostname.
+
+.. _delegate_parallel:
+
+Parallelism
+-----------
```
Comment (acozine) [inline docs/docsite/rst/user_guide/playbooks_delegation.rst]:
```suggestion
Delegation and parallel execution
---------------------------------
```

**Review by acozine — COMMENTED** [review]
Adding a few more suggestions.

File: docs/docsite/rst/user_guide/playbooks_delegation.rst [hunk 9]
```diff
@@ -77,7 +77,24 @@ To specify more arguments, use the following syntax::
...
+Most commonly users are affected by this when updating a single file on a single delegated to host for all hosts (via copy/template/lineinfile/etc), they will still operate in parallel forks (default 5) and overwrite each other.
+
+This can be handled in several ways::
+
+    - name: "handle concurrency via a loop on the hosts with `run_once: true`"
+      lineinfile: "<options here>"
+      run_once: true
+      loop: '{{ ansible_play_hosts_all }}'
+
+By using an intermediate play with  `serial: 1` or using  `throttle: 1` at task level.
```
Comment (acozine) [inline docs/docsite/rst/user_guide/playbooks_delegation.rst]: Another example would be good here, and/or link to https://docs.ansible.com/ansible/latest/user_guide/playbooks_strategies.html#setting-the-batch-size-with-serial and https://docs.ansible.com/ansible/latest/user_guide/playbooks_strategies.html#restricting-execution-with-throttle.

**Review by samccann — APPROVED** [review]

**Pull request merged**